packages/blocks/src/api/registration.js

```diff
@@ -373,3 +373,22 @@ export const registerBlockStyle = ( blockName, styleVariation ) => {
 		};
 	} );
 };
+
+/**
+ * Unregisters a block style variation for the given block.
+ *
+ * @param {string} blockName          Name of block (example: “core/latest-posts”).
+ * @param {string} styleVariationName Name of class applied to the block.
+ */
+export const unregisterBlockStyle = ( blockName, styleVariationName ) => {
+	addFilter( 'blocks.registerBlockType', `${ blockName }/${ styleVariationName }/unregister`, ( settings, name ) => {
```

Comment:
Maybe we have to rethink the namespace for this and `registerBlockStyle`.

Reply:
FWIW, I'd personally like to see if we could avoid the filter altogether, and instead make style a proper data type of its own than force-fit it into the block definition.

Reply:
It would be nice to avoid using the filter in the first place.

In regards to the propose PR, why not use buil-int `removeFilter` instead?
```js
removeFilter( 'blocks.registerBlockType', `${ blockName }/${ styleVariation.name }` );
```

Reply:
It would be nice to avoid using the filter in the first place.

In regards to the propose PR, why not use buil-int `removeFilter` instead?
```js
removeFilter( 'blocks.registerBlockType', `${ blockName }/${ styleVariation.name }` );
```

Reply:
It would be nice to avoid using the filter in the first place.

In regards to the propose PR, why not use buil-int `removeFilter` instead?
```js
removeFilter( 'blocks.registerBlockType', `${ blockName }/${ styleVariation.name }` );
```

Reply:
@gziolo Because you can't remove built-in style variations with this.